tests/by-util/test_touch.rs

```diff
@@ -118,6 +119,47 @@ fn test_touch_set_mdhms_time() {
     assert_eq!(mtime.unix_seconds() - start_of_year.unix_seconds(), 45296);
 }
 
+#[test]
+fn test_y2038_touch() {
+    // Jan 19 2038, 03:14:07 UTC is the moment of the Y2038 bug
+    // test 1 second before, at, and 1 second after
+    let year = 2038;
+    let month = 1;
+    let day = 19;
+    let hours = 3;
+    let minutes = 14;
+    let seconds = 7;
+    for deltas in [-1i32, 0, 1] {
+        let (at, mut ucmd) = at_and_ucmd!();
+
+        let date = NaiveDate::from_ymd_opt(year, month, day).unwrap();
+        let datetime = date
+            .and_hms_opt(hours, minutes, (seconds + deltas).try_into().unwrap())
+            .unwrap();
+        let timestamp = datetime.and_utc().timestamp();
+
+        let date = format!(
+            "{}{:02}{:02}{:02}{:02}.{:02}",
+            year,
+            month,
+            day,
+            hours,
+            minutes,
+            seconds + deltas
+        );
+        let file = format!("Y2038_{}_{}", date, deltas);
+        ucmd.args(&["-t", &date, &file]).succeeds().no_output();
```

Comment:
Maybe it would be simpler/more concise to just specify the time argument directly? So the three test cases would be

```
ucmd.args(&["-t", "203801190314.06", "f"])
ucmd.args(&["-t", "203801190314.07", "f"])
ucmd.args(&["-t", "203801190314.08", "f"])
```

Reply:
I expanded to 3 tests one just before, one at and one just after.